torch_geometric/data/graph_store.py

```diff
@@ -92,15 +116,134 @@ def get_edge_index(self, *args, **kwargs) -> EdgeTensorType:
         """
         edge_attr = self._edge_attr_cls.cast(*args, **kwargs)
         edge_attr.layout = EdgeLayout(edge_attr.layout)
+        # Override is_sorted for CSC and CSR:
+        edge_attr.is_sorted = edge_attr.is_sorted or (edge_attr.layout in [
+            EdgeLayout.CSC, EdgeLayout.CSR
+        ])
         edge_index = self._get_edge_index(edge_attr)
         if edge_index is None:
             raise KeyError(f"An edge corresponding to '{edge_attr}' was not "
                            f"found")
         return edge_index
 
-    # TODO implement coo(), csc(), csr() methods on GraphStore, which perform
-    # conversions of edge indices between formats. These conversions can also
-    # automatically be performed in `get_edge_index`
+    def _to_layout(self, to_layout: EdgeLayout,
+                   store: bool = False) -> ConversionOutputType:
+        # Obtain all edge attributes, grouped by type:
+        edge_attrs = self.get_all_edge_attrs()
+        edge_type_to_attrs: Dict[Any, List[EdgeAttr]] = defaultdict(list)
+        for attr in edge_attrs:
+            edge_type_to_attrs[attr.edge_type].append(attr)
+
+        # Convert layouts for each attribute from its most favorable original
+        # layout to the desired layout. Store permutations of edges if
+        # necessary as part of the conversion:
+        row_dict, col_dict, perm_dict = {}, {}, {}
+        for _, edge_attrs in edge_type_to_attrs.items():
+            edge_layouts = [edge_attr.layout for edge_attr in edge_attrs]
+
+            # Ignore if requested layout is already present:
+            if to_layout in edge_layouts:
+                from_attr = edge_attrs[edge_layouts.index(to_layout)]
+                row, col = self.get_edge_index(from_attr)
+                perm = None
+
+            # Convert otherwise:
+            else:
+                # Pick the most favorable layout to convert from. We always
+                # prefer CSR or CSC to COO:
```

Comment:
There is no real reason to do that IMO. COO -> CSC is actually preferred over CSR -> CSC because you do not have to unwrap the compressed representation.

Reply:
Gotcha, changed - thanks!